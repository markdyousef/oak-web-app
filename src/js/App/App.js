import React, { Component, PropTypes } from 'react';
import Sidebar from './Sidebar.js';
import Index from './AppIndex.js';
import css from './app.css';

const styles = {
    height: '100%',
    background: '#333'
};

class App extends Component {
    static propTypes = {
        content: PropTypes.object,
        sidebar: PropTypes.object
    }

    constructor() {
        super();
        this.state = {};
    }

    render() {
        const { content, sidebar } = this.props;

        return (
            <div style={styles} className={css.container}>
                <h1 className="welcome-header">Welcome to testing React!</h1>
                <div className="Sidebar">
                    {sidebar || <Sidebar onMount={() => {}} isActive />}
                </div>
                <div className="Content">
                    {content || <Index />}
                </div>
            </div>
        );
    }
}

export default App;
