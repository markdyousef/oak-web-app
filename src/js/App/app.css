body {
  font-family: 'Helvetica';
  background: #ddd;
}

.container {
  width: 100%;
  border-radius: 15px;
  transform: rotate(1deg);
}
