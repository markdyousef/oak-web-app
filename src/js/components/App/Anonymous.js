// @flow
import React, { Component, PropTypes } from 'react';
import css from './App.css';


// Localstorage token
const token = localStorage.authToken;

class Anonymous extends Component {
    static propTypes = {
        children: PropTypes.node.isRequired,
        location: PropTypes.shape({
            pathname: PropTypes.string
        }).isRequired,
        router: PropTypes.shape({
            replace: PropTypes.func.isRequired
        }).isRequired
    };
    constructor() {
        super();
        this.state = {};
    }
    componentWillMount() {
        const { router, location } = this.props;
        if (token && token.length > 0) {
            router.replace({
                pathname: '/',
                state: { nextPathname: location.pathname }
            });
        }
    }
    render() {
        console.log(this.props.children);
        if (token) return null;

        return (
            <div className={css.container}>
                {this.props.children}
            </div>
        );
    }
}

export default Anonymous;
