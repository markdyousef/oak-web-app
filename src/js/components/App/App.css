.container {
    width: 100%;
    height: 100%;
    display: flex;
    position: absolute;
}

.right {
    width: 100%;
}
