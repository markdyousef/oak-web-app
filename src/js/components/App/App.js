import React, { Component, PropTypes } from 'react';
import TopNav from '../TopNav';
import SideNav from '../SideNav';

import css from './App.css';


class App extends Component {
    static propTypes = {
        children: PropTypes.node
    }

    constructor() {
        super();
        this.state = {};
    }

    render() {
        return (
            <div className={css.container}>
                <SideNav />
                <div className={css.right}>
                    <TopNav />
                    {this.props.children}
                </div>
            </div>
        );
    }
}

export default App;
