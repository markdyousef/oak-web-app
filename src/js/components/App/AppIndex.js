import React, { Component } from 'react';


class Index extends Component {
    constructor() {
        super();
        this.state = {};
    }

    render() {
        return (
            <div>
                <h1>Sidebar</h1>
                <p>
                    Routes can have multiple components, so that all portions of your UI
                    can participate in the routing.
                </p>
            </div>
        );
    }
}


export default Index;
