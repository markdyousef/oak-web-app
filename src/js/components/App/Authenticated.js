// @flow
import React, { Component, PropTypes } from 'react';
import SideNav from '../../containers/SideNavContainer';

import css from './App.css';

// Localstorage token
const token = localStorage.authToken;

class Authenticated extends Component {
    static propTypes = {
        children: PropTypes.node.isRequired,
        router: PropTypes.shape({
            replace: PropTypes.func.isRequired
        }).isRequired
    };
    constructor() {
        super();
        this.state = {};
    }
    componentWillMount() {
        const { router } = this.props;
        if (!token) {
            router.replace({
                pathname: '/login'
            });
        }
    }
    render() {
        if (!token) return null;

        return (
            <div className={css.container}>
                <SideNav />
                <div className={css.right}>
                    {this.props.children}
                </div>
            </div>
        );
    }
}

export default Authenticated;
