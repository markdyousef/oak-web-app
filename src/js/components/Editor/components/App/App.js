import React, { Component } from 'react';
import { EditorState, RichUtils, Editor } from 'draft-js';
// TODO: add custom styling
import 'draft-js/dist/Draft.css';
import styled from 'styled-components';
import Toolbar from '../Toolbar';
import FloatingActionButton from '../FloatingActionButton';
import customRenderer from '../../utils/customRenderer';
import { Block } from '../../utils/constants';

const Container = styled.div`
    ${''/* padding: 30px 30px; */}
    box-sizing: border-box;
    position: relative;
    & h1 {
        font-size: 48px;
        font-weight: bold;
        letter-spacing: -.024em;
        line-height: 1.18;
        margin-bottom: 20px;
        color: #131517;

    }
    & h2 {
        font-size: 28px;
        font-weight: normal;
        letter-spacing: -.008em;
        line-height: 1.24;
        margin-bottom: 20px;
        color: #797C80;
    }
    & ul {
        padding-left: 0;
        list-style: none;
    }
    & ol {
        padding-left: 0;
        list-style: none;
    }
    & li {
        font-size: 21px;
        line-height: 1,78;
    }
`;

const EditorContainer = styled.div`
    cursor: text;
    position: relative;
    margin: 0 auto;
    margin-top: 10px;
    max-width: 700px;
`;

export default class App extends Component {
    constructor() {
        super();
        this.state = {
            editorState: EditorState.createEmpty()
        };
    }
    onChange = editorState => this.setState({ editorState })
    onTab = (event) => {
        const { editorState } = this.state;
        // depth on ul and ol
        const levels = 2;
        const newEditorState = RichUtils.onTab(event, editorState, levels);
        if (newEditorState !== editorState) {
            this.onChange(newEditorState);
        }
    };
    getEditorState = () => this.state.editorState;
    focus = () => this.editor.focus();
    toggleBlockType = (blockType) => {
        const { editorState } = this.state;
        const type = RichUtils.getCurrentBlockType(editorState);
        if (type.indexOf(`${Block.ATOMIC}:`) === 0) return;
        this.onChange(
            RichUtils.toggleBlockType(
                editorState,
                blockType
            )
        );
    }
    toggleInlineStyle = (inlineStyle) => {
        const { editorState } = this.state;
        this.onChange(
            RichUtils.toggleInlineStyle(
                editorState,
                inlineStyle
            )
        );
    }
    render() {
        const { editorState } = this.state;
        const showToolbar = !editorState.getSelection().isCollapsed();
        return (
            <Container>
                <EditorContainer>
                    <Editor
                        ref={(node) => { this.editor = node; }}
                        editorState={editorState}
                        spellCheck
                        placeholder="Write something cool..."
                        onChange={this.onChange}
                        blockRendererFn={customRenderer(editorState, this.onChange)}
                        onTab={this.onTab}
                    />
                    <FloatingActionButton
                        editorState={editorState}
                        focus={this.focus}
                        setEditorState={this.onChange}
                    />
                    <Toolbar
                        editorState={editorState}
                        toggleBlockType={this.toggleBlockType}
                        toggleInlineStyle={this.toggleInlineStyle}
                        focus={this.focus}
                        editorNode={this.editor}
                        showToolbar={showToolbar}
                    />
                </EditorContainer>
            </Container>
        );
    }
}
