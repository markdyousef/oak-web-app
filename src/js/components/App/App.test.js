/* eslint-disable no-undef */
import React from 'react';
// import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import App from './App';

// const wrapper = shallow(
//     <App />
// );

it('renders correctly', () => {
    const component = renderer.create(
        <App />
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});

// it('renders as a <div>', () => {
//     expect(wrapper.type()).to.eql('div');
// });
