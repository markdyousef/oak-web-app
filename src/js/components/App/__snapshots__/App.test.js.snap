exports[`test renders correctly 1`] = `
<div
  className={undefined}>
  <div
    className={undefined}>
    <div
      className={undefined}>
      <a
        onClick={[Function]}
        style={Object {}}>
        <div />
        <h1>
          Collections
        </h1>
      </a>
      <div
        className={undefined}>
        <div>
          <a
            onClick={[Function]}
            style={Object {}}>
            Design
          </a>
        </div>
        <div>
          <a
            onClick={[Function]}
            style={Object {}}>
            Frontend
          </a>
        </div>
        <div>
          <a
            onClick={[Function]}
            style={Object {}}>
            Strategy
          </a>
        </div>
      </div>
    </div>
    <button
      className=" "
      onClick={[Function]}>
      <div>
        <label
          className={undefined}
          htmlFor="text">
          Add Collection
        </label>
      </div>
    </button>
  </div>
  <div
    className={undefined}>
    <nav
      className={undefined}>
      <div
        className={undefined} />
      <div
        className={undefined}>
        <a
          className={undefined}
          onClick={[Function]}
          style={Object {}}>
          <div
            className={undefined}>
            <img
              alt="profile"
              src="//style.anu.edu.au/_anu/4/images/placeholders/person.png" />
          </div>
        </a>
      </div>
    </nav>
  </div>
</div>
`;
